LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs

```diff
@@ -134,8 +136,7 @@ public byte[] Serialize(AppKey appKey)
         {
             Mic = LoRaWan.Mic.ComputeForJoinAccept(appKey, MHdr, AppNonce, NetId, DevAddr, DlSettings, RxDelay, CfList);
             _ = PerformEncryption(appKey);
-
-            return GetByteMessage();
+            return this.rawMessage.Prepend((byte)MHdr).ToArray();
```

Comment:
Aren't we prepending the `MHdr` twice now?

Reply:
I don't have a good answer for this right now as I haven't looked at the `rawMessage` preparation in detail. All that was done here was inlining of `GetByteMessage` where this was the sole remaining call, and `GetByteMessage` was always prepending the header byte. However, if we consider than it's being set as the first byte of `rawMessage` on line 54 (like before) then it does seem like a duplication.

Reply:
So I took a closer look as I was wondering how all the tests were still passing and turns out that this worked because `PerformEncryption` rewrites `rawMessage` without the header and this adds it back. So it turns out, what read like a duplicate addition of the header byte, isn't actually one. This also means that the `rawMessage` setup in the constructor was never used since it was always overwritten by `PerformEncryption`. I've completely removed the need for `rawMessage` with a97c56b53ca8cb4c4a51098997d3defa93216eed. It's a more dramatic change than I was hoping to make here, but I think it makes sense to go with it as it reads cleaner (the return value from `PerformEncryption` also doesn't get discarded anymore. I am also tempted to inline the single use of `PerformEncryption` into `Serialize`. Thoughts?

Reply:
By the way, just realized that the `rawMessage` setup in the constructor was never needed or used anymore so completely eliminated it with f5a2217e2d4e86c1ff87f3c48552b3983ea24085.

Reply:
> I am also tempted to inline the single use of `PerformEncryption` into `Serialize`.

Done with 9d9a0d211e8e2881474ed78443e4f43671b015c3 but can be reverted if you're not in favour of this.